writeengine/bulk/we_colextinf.cpp

```diff
@@ -206,68 +209,80 @@ void ColExtInf::getCPInfoForBRM(JobColumn column, BRMReporter& brmReporter)
     int128_t bigMinVal = iter->second.fbigMinVal;
     int128_t bigMaxVal = iter->second.fbigMaxVal;
 
-    if (bIsChar)
+    bool bIsValid = true;
+
+    if (bIsChar || bIsText)
     {
       // If we have added 1 or more rows, then we should have a valid
       // range in our RowExtMap object, in which case...
       // We swap/restore byte order before sending min/max string to BRM;
       // else we leave fMinVal & fMaxVal set to LLONG_MIN and send as-is,
       // to let BRM know we added no rows.
+      
       if ((iter->second.fMinVal != iter->second.fMaxVal) || (iter->second.fMinVal != LLONG_MIN))
       {
         minVal = static_cast<int64_t>(uint64ToStr(static_cast<uint64_t>(iter->second.fMinVal)));
         maxVal = static_cast<int64_t>(uint64ToStr(static_cast<uint64_t>(iter->second.fMaxVal)));
       }
+      else
+      {
+        // This is dropping range to invalid.
```

Comment:
cpimport drops ranges for extents it writes. Then it computes these (in case of character columns incorectly, in my opinion - LLONG_MIN???) and tries to set these.

The setting of the min/max values is from the abundance of caution. The lowering of bIsValid is needed, so that we do not set incorrect ranges. The BRM call that cpimport uses does not check validity of range and just assigns CP_VALID there, making empty arnge valid and make all extent elimination oppeations to succeed.

Thus, bIsValid==false means we will not pass the range at all, leaving arnge in CP_INVALID and triggeing a scan.

Reply:
Much appreciated